### ISSUE TITLE
Low Dice Scores on NETC and SNFH classes despite good performance on WT/ET (using real training data only)

### ISSUE THREAD

**Issue by Pilmaut** [ShadowTwin41/BraTS_2023_2024_solutions#8]
Hi,

First of all, thanks a lot for sharing this codebase 🙏. It's very useful to start a project and try to reproduce brats challenge. 
I have tested the pretrained weights shared in the README for 2024. 

For inference, I strictly followed the provided pipeline and did not train any new model.

On Synapse evaluation 2025 plateform (with n=219 cases), I obtained the following results:

Metric | Mean
Dice WT | 0.85
Dice ET | 0.79
Dice TC | 0.39
Dice NETC | 0.12
Dice SNFH | 0.005
HD95 WT | 22.5 mm
HD95 ET | 19.9 mm
HD95 TC | 37.3 mm
HD95 NETC | 44.3 mm
HD95 SNFH | 102.5 mm

So while the model performs well on WT and ET, the performance on NETC and SNFH is extremely low.

My questions are described as follow: 

1. Are the released pretrained weights expected to perform this way (good WT/ET, but very low NETC/SNFH)?

2. Is there a recommended label remapping or post-processing step (e.g. 4→3 relabeling) that could impact NETC/SNFH metrics?

3. For Synapse submission: is running 

>>BraTS2024_inference/main.py file enough or is there any additional step required before packaging and submitting the results ? 

Thanks a lot for your help and clarification :) 

Pilmaut

**Comment by ShadowTwin41** [comment]
Hello, thanks for reaching out!

1. No, the model was expected to perform well on all metrics, as you can see in https://www.synapse.org/Synapse:syn51156910/wiki/622343 submissions of Faking_it team. Therefore, I assume something is wrong in the pre-processing. 

2. Please, take a look at https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/dataset_conversion/Dataset137_BraTS21.py as I use the same method as them.

3. Yes that's the correct file. However, the solution from 2024 is for the Adult Glioma Post Treatment challenge. Which means that it expects the resection cavity (RC - label 4). I suggest using the solution from 2023 for pre-op. In any case, the solution from 2024 should work better than the results you presented. Therefore, I suspect there is a problem with the labels/pre-processing. 

Let me know if you have further questions!
